app/Http/Controllers/Articles/ArticlesController.php

```diff
@@ -101,11 +103,13 @@ public function create()
 
     public function store(ArticleRequest $request)
     {
-        $article = $this->dispatchNow(CreateArticle::fromRequest($request));
+        $article = $this->dispatchSync(CreateArticle::fromRequest($request));
 
         $this->success($request->shouldBeSubmitted() ? 'articles.submitted' : 'articles.created');
 
-        return redirect()->route('articles.show', $article->slug());
+        return $request->expectsJson()
```

Comment:
Is it possible to choose either `expectsJson` or `wantsJson`. AFAIK, either will work, but `wantsJson` seems the better choice as `expectsJson` looks for an ajax request or the presence of json in the `Accept` header, where `wantsJson` just the latter which feels like the right choice to me. I might be missing something though.

Reply:
Good catch. Moved to `wantsJson`